### PR TITLE
With DEBUG_LEAKS on: Print the initialize callstack of active async resources which may be preventing node from shutting down 

### PR DESCRIPTION
**Pull request by nick4598**
Also removes unnecessary double clone from map-layers-format extension.

### CODE REVIEW

File: tools/build/src/mocha-reporter/index.ts [hunk 1]
```diff
@@ -6,12 +6,33 @@
...
+      return;
+    }
+    asyncResourceStats.delete(asyncId);
+  };
 
+  const asyncHook = async_hooks.createHook({init, destroy});
+  asyncHook.enable();
+}
 const fs = require("fs-extra");
+import * as path from "path";
```
Comment (aruniverse) [inline tools/build/src/mocha-reporter/index.ts:35]: feels weird using both import and require, can we consolidate and just use require here?

File: tools/build/src/mocha-reporter/index.ts [hunk 2]
```diff
@@ -76,6 +97,17 @@ class BentleyMochaReporter extends Spec {
           const wtf = require("wtfnode");
           wtf.setLogger("info", console.error);
           wtf.dump();
+
+          let activeResourcesInfo: string[] = (process as any).getActiveResourcesInfo(); // https://nodejs.dev/en/api/v18/process#processgetactiveresourcesinfo (Not added to @types/node yet I suppose)
```
Comment (aruniverse) [inline tools/build/src/mocha-reporter/index.ts:101]: youve tried updating to the latest @types/node@18.x ver?

**Review by aruniverse — APPROVED** [review]
great work!

File: tools/build/src/mocha-reporter/index.ts (hunk 2 above)
Comment (nick4598) [reply tools/build/src/mocha-reporter/index.ts:101]: No I didn't, but I just checked definitelyTyped repo and I don't see it mentioned anywhere except for a couple of comments in async_hooks types file.

File: tools/build/src/mocha-reporter/index.ts (hunk 1 above)
Comment (nick4598) [reply tools/build/src/mocha-reporter/index.ts:35]: Changed it

**Review by pmconne — APPROVED** [review]

**Pull request merged**